sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -1474,17 +1438,11 @@ class MultiFileCloudOrcPartitionReader(
         val batchSizeBytes = GpuColumnVector.getTotalDeviceMemoryUsed(table)
         logDebug(s"GPU batch size: $batchSizeBytes bytes")
         maxDeviceMemory = max(batchSizeBytes, maxDeviceMemory)
-        val numColumns = table.getNumberOfColumns
-        if (readDataSchema.length != numColumns) {
-          throw new QueryExecutionException(s"Expected ${readDataSchema.length} columns " +
```

Comment:
Looks like the similar exception is not thrown in the  evolveSchemaIfNeededAndClose in the production version.

Reply:
Yes, it is removed intentionally, because it is no longer correct after supporting the feature mentioned in issue #3058.
The final output batch consists of two types of columns. Some are read from ORC files, and the other (columns filled with nulls) are generated by our plugin, to align with the behavior of Spark ORC reader.